client/cypress/integration/dashboard/dashboard_spec.js

```diff
@@ -395,6 +523,71 @@ describe('Dashboard', () => {
             .should('eq', 335);
         });
       });
+
+      describe('adjusts height on refresh', () => {
+        const paramName = 'count';
+        const queryData = {
+          query: `select s.a FROM generate_series(1,{{ ${paramName} }}) AS s(a)`,
+        };
+
+        beforeEach(function () {
+          addWidgetByAPI(this.dashboardId, queryData).then((elTestId) => {
+            cy.visit(this.dashboardUrl);
+            cy.getByTestId(elTestId).as('widget').within(() => {
+              cy.getByTestId('RefreshIndicator').as('refreshButton');
+            });
+            cy.getByTestId(`ParameterName${paramName}`).within(() => {
+              cy.get('input').as('paramInput');
+            });
+          });
+        });
+
+        it('grows when dynamically adding table rows', () => {
+          // listen to results
+          cy.server();
+          cy.route('GET', 'api/query_results/*').as('FreshResults');
+
+          // start with 1 table row
+          cy.get('@paramInput').clear().type('1');
+          cy.get('@refreshButton').click();
+          cy.wait('@FreshResults', { timeout: 10000 });
+          cy.get('@widget').invoke('height').should('eq', 285);
+
+          // add 4 table rows
+          cy.get('@paramInput').clear().type('5');
+          cy.get('@refreshButton').click();
+          cy.wait('@FreshResults', { timeout: 10000 });
+
+          // expect to height to grow by 1 grid grow
+          cy.get('@widget').invoke('height').should('eq', 435);
+        });
+
+        it('auto height revoked after manual height adjustment', () => {
+          // listen to results
+          cy.server();
+          cy.route('GET', 'api/query_results/*').as('FreshResults');
+
+          editDashboard();
+
+          // start with 1 table row
+          cy.get('@paramInput').clear().type('1');
+          cy.get('@refreshButton').click();
+          cy.wait('@FreshResults');
+          cy.get('@widget').invoke('height').should('eq', 285);
+
+          // resize height by 1 grid row
+          resizeBy(cy.get('@widget'), 5);
+          cy.get('@widget').invoke('height').should('eq', 335);
+
+          // add 4 table rows
+          cy.get('@paramInput').clear().type('5');
+          cy.get('@refreshButton').click();
+          cy.wait('@FreshResults');
+
+          // expect height to stay unchanged (would have been 435)
```

Comment:
Just to show a different way to assert the same thing:
```javascript
expect(() => {
  // add 4 table rows
  cy.get('@paramInput').clear().type('5');
  cy.get('@refreshButton').click();
  cy.wait('@FreshResults');
}).not.to.change(this.widget, 'height');
```

Reply:
Oooh that's nice.

Reply:
But I don't think this specifically would work cause
1. `height` is a method, not a property.
2. `this.widget` is a cy object, not jquery.

Right?

Reply:
Ideally the `dragBy` and `resizeBy` functions would not return a delta value but instead:
```js
it('moves one column', () => {
  const getLeft = () => $el.offset().left;
  const drag = () => dragBy($el, 200);

  expect(drag).to.increase(getLeft).by(200);
});
```
Then `resizeBy` would simply be a `dragBy` of the element's handle. So nice.

BUT, disconnecting the drag from the delta calculation is currently problematic since the dragged element animates to its new position on mouseup (compensated by calculating the drag placeholder position instead).
Any workaround suggestion? (wait for animation end, disable animation for test, ...)

Reply:
Really cool, once I get to my laptop I'll run a few tests with it. For the questions:

> But I don't think this specifically would work cause

It worked for success, need to try on failure haha. The assertion is essentially the same, they just run it before the function and then guarantee it's the same after.

> height is a method, not a property

I hope not to have seen the above wrongly, but this is not supposed to be a problem

> this.widget is a cy object, not jquery.

Need to confirm, but I think it is a jQuery object, when I saw it in docs it seemed the same as getting the result of `cy.get('@widget')`

Reply:
If you can make it work that would be awesome 👍

Reply:
I could not :sweat_smile:. That works well with javascript objects, not with elements (I could use a helper object to represent the element, but I don't think it is worth it for my example).

Yours is much better, it would make tests a lot more readable.

> Any workaround suggestion?

Can we use the placeholder position for the `getLeft` method? If not, NP, we can move on.

Reply:
> Can we use the placeholder position for the getLeft method? If not, NP, we can move on.

It doesn't get rendered in the dom until mousedown (or mousemove, I didn't check) so it's not ideal.